### ISSUE TITLE
wlr/taskbar: Items aren't consistently ordered

### ISSUE THREAD

**Issue by iguanajuice** [Alexays/Waybar#2559]
The taskbar module seems to order by "oldest first" by default. Even with `"sort-by-app-id": "true"`, it will still sort this way for windows that share the same app_id.
However, if you reload waybar or turn your display fully off and then on again, the ordering switches to "newest first". Why the inconsistency? It's a small detail, but it's driving me nuts.

**Comment by iguanajuice** [comment]
I tried fixing this issue myself, but I only ran into something very strange. I edited `src/modules/wlr/taskbar.cpp` and I changed
```
void Taskbar::handle_toplevel_create(struct zwlr_foreign_toplevel_handle_v1 *tl_handle) {
  tasks_.push_back(std::make_unique<Task>(bar_, config_, this, tl_handle, seat_));
}
```
to this
```
void Taskbar::handle_toplevel_create(struct zwlr_foreign_toplevel_handle_v1 *tl_handle) {
  tasks_.insert(tasks_.begin(), std::make_unique<Task>(bar_, config_, this, tl_handle, seat_));
}
```
This makes it where new `tasks_` are placed first, since the bug I'm experiencing makes tasks reorder to "newest first", making them spawn this way would make their position consistent.
**However.** When I build and run waybar with these changes, the inconsistency somehow persists.

In short, if new tasks are made to be ordered first, the bug reorders them to be ordered last.
For some reason, it always does the opposite. I have no clue how this is possible.

**Comment by headbank** [comment]
I hope this is not unwelcome noise here, but it seems worth mentioning here instead of (or as preparation for) filing separately. My config is only this:
```
    "wlr/taskbar": {
       	"on-click": "minimize-raise"
    }
```
What I find is that ordering between other apps stays oldest-first, but among a group of the same app but with different app-ids (in my case multiple of footclient), the order seems to shuffle randomly over time. At login the order was (in order of launching) 0,1,2,3 but now it is 2,1,0,3. I can't confirm more than one shuffle per session (yet) and can't see any explanation for the ordering it ends up with.

For background, I'm running on labwc headless over wayvnc, on Gentoo. The footclient instances are autostarted with the session, each with a different app-id defined on the commandline so that I can use labwc's window rules to position them in a grid:
```
for n in 1 2 3 4; do
    footclient -a footclient-quad$n -w 600x350 >/dev/null 2>&1 &
    sleep 0.5
done
```
[The `sleep` is because when I first noticed this issue I suspected some kind of race happening at start. Since adding it the ordering does appear to be correct at start, but it doesn't stay that way.]

Happy to post this separately if it's deemed a separate issue.
